### ISSUE TITLE
Don't assume paths are strings

### ISSUE THREAD

**Issue by alex000kim** [iterative/mlem#359]
Some parts of the code assume paths to be strings e.g. https://github.com/iterative/mlem/blob/4666de893967cdde41bf5ff605acf51a84bc9b68/mlem/utils/path.py#L8

If you pass a `pathlib`'s path, then you'd get the following error:
```
...
      6 if not path:
      7     return path
----> 8 return path.replace("\\", "/")

TypeError: replace() takes 2 positional arguments but 3 were given
```

**Comment by madhur-tandon** [comment]
Is this really a bug though? Because the function signature clearly has `make_posix(path: Optional[str])` i.e. `Optional[str]`, so not really sure if this should be handled or not.
CC: @mike0sv

**Comment by aguschin** [comment]
Yep, I guess it's not a bug, but an improvement we could implement. Let me update the label...
